test/jdk/javax/swing/JButton/HtmlButtonImageTest/HtmlButtonImageTest.java

```diff
@@ -0,0 +1,155 @@
+            robot.setAutoWaitForIdle(true);
+
+            // store path to source directory to locate image
+            testDir = Path.of(System.getProperty("test.classes", "."));
+
+            // generate red_square.png image to use as JButton text
+            generateImage();
+
+            SwingUtilities.invokeAndWait(() -> createAndShowGUI());
+
+            robot.waitForIdle();
+
+            // retrieve color of pixels at each edge of square image by starting at the center of the button
+            setupCenterCoord();
+            robot.mouseMove(point.x, point.y);
+
+            // store each pixel color on the edge of each side of the red square
+            Color leftClr = robot.getPixelColor(point.x - (SQUARE_WIDTH/2) + PIXEL_BUFFER, point.y);
+            Color rightClr = robot.getPixelColor(point.x + (SQUARE_WIDTH/2) - PIXEL_BUFFER, point.y);
+            Color topClr = robot.getPixelColor(point.x, point.y - (SQUARE_HEIGHT/2) + PIXEL_BUFFER);
+            Color botClr = robot.getPixelColor(point.x, point.y + (SQUARE_HEIGHT/2) - PIXEL_BUFFER);
+
+            testImageCentering(leftClr, rightClr, topClr, botClr);
+        } finally {
+            // dispose frame when done testing for a LAF before continuing
+            SwingUtilities.invokeAndWait(() -> frame.dispose());
+        }
+    }
+
+    private static void createAndShowGUI()
+    {
+        frame = new JFrame();
+        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        frame.setLayout(new FlowLayout());
+
+        // create JButton of size 37x37 text set to a 19x19 image of a red square loaded through html tags
+        button = new JButton();
+        button.setFocusPainted(false);
+        button.setPreferredSize(new Dimension(BUTTON_WIDTH, BUTTON_HEIGHT));
+        button.setText("<html><img src='" + testDir.resolve("red_square.png").toUri() + "'></html>");
+
+        frame.add(button);
+        frame.pack();
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }
+
+    private static void setupCenterCoord() {
+        // adjust coordinates to be the center of the button
+        point = button.getLocationOnScreen();
+        point.x += BUTTON_WIDTH / 2;
+        point.y += BUTTON_HEIGHT / 2;
+    }
+
+    private static boolean checkRedness(Color c) {
+        // checks for redness since anti-aliasing causes edges to be not exactly 255,0,0 rgb values
+        if(c.getRed() > 250 && c.getBlue() < 10 && c.getGreen() < 10) {
+            return true;
+        }
+        return false;
+    }
+
+    private static void testImageCentering(Color left, Color right, Color top, Color bottom) {
+        // check if all colors at each edge of square are red
+        if(!checkRedness(left) || !checkRedness(right) || !checkRedness(top) || !checkRedness(bottom)) {
+            throw new RuntimeException("HTML image not centered in button");
+        }
+        else {
+            System.out.println("-- Passed");
+        }
+    }
```

Comment:
I suggest using variable arguments:
```suggestion
    private static void testImageCentering(Color... colors) {
        // check if all colors at each edge of square are red
        for (Color c : colors) {
            if (!checkRedness(c)) {
                throw new RuntimeException("HTML image not centered in button");
            }
        }
       System.out.println("-- Passed");
    }
```
The call to this function remains the same. And you can easily add additional points to be checked if needed.

Reply:
Awesome idea, added.